commit d0f7fbdb1cf98e37b3c61b503d41578517cede1c
Author: Stephen Boyd <swboyd@chromium.org>
Date:   Thu Jun 21 19:27:16 2018 -0700

    HID: i2c-hid: Use devm to allocate i2c_hid struct
    
    [ Upstream commit d6f83894110de247a81392ab7ef89e5498df7e80 ]
    
    Use devm here to save some lines and prepare for bulk regulator usage in
    this driver. Otherwise, when we devm bulk get regulators we'll free the
    containing i2c_hid structure and try to put regulator pointers from
    freed memory.
    
    Cc: Benjamin Tissoires <benjamin.tissoires@redhat.com>
    Cc: Hans de Goede <hdegoede@redhat.com>
    Cc: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Cc: Dmitry Torokhov <dtor@chromium.org>
    Cc: Doug Anderson <dianders@chromium.org>
    Signed-off-by: Stephen Boyd <swboyd@chromium.org>
    Acked-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
    Signed-off-by: Jiri Kosina <jkosina@suse.cz>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/hid/i2c-hid/i2c-hid.c b/drivers/hid/i2c-hid/i2c-hid.c
index 5fd1159fc095..64773433b947 100644
--- a/drivers/hid/i2c-hid/i2c-hid.c
+++ b/drivers/hid/i2c-hid/i2c-hid.c
@@ -1004,18 +1004,18 @@ static int i2c_hid_probe(struct i2c_client *client,
 		return client->irq;
 	}
 
-	ihid = kzalloc(sizeof(struct i2c_hid), GFP_KERNEL);
+	ihid = devm_kzalloc(&client->dev, sizeof(*ihid), GFP_KERNEL);
 	if (!ihid)
 		return -ENOMEM;
 
 	if (client->dev.of_node) {
 		ret = i2c_hid_of_probe(client, &ihid->pdata);
 		if (ret)
-			goto err;
+			return ret;
 	} else if (!platform_data) {
 		ret = i2c_hid_acpi_pdata(client, &ihid->pdata);
 		if (ret)
-			goto err;
+			return ret;
 	} else {
 		ihid->pdata = *platform_data;
 	}
@@ -1128,7 +1128,6 @@ static int i2c_hid_probe(struct i2c_client *client,
 
 err:
 	i2c_hid_free_buffers(ihid);
-	kfree(ihid);
 	return ret;
 }
 
@@ -1152,8 +1151,6 @@ static int i2c_hid_remove(struct i2c_client *client)
 
 	regulator_disable(ihid->pdata.supply);
 
-	kfree(ihid);
-
 	return 0;
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1188_d0f7fbdb1cf98e37b3c61b503d41578517cede1c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1188_d0f7fbdb1cf98e37b3c61b503d41578517cede1c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
